components/board/genericlinux/pin_types.go

```diff
@@ -41,3 +33,61 @@ type NoBoardFoundError struct {
 func (err NoBoardFoundError) Error() string {
 	return fmt.Sprintf("could not determine %q model", err.modelName)
 }
+
+// PinDefinition describes a gpio pin on a linux board.
+type PinDefinition struct {
+	Name            string `json:"name"`
+	Ngpio           int    `json:"ngpio"` // this is the ngpio number of the chip the pin is attached to
+	LineNumber      int    `json:"line_number"`
```

Comment:
Worth putting in a comment saying this is a relative line number on the chip, rather than a global line number? Maybe not; maybe keep as-is.